### DISCUSSION TITLE
New documentation theme

### DISCUSSION THREAD

**Discussion by dantownsend** [piccolo-orm/piccolo#320] [Ideas]
I've been looking at alternative themes for Sphinx.

[Sphinx Material](https://github.com/bashtage/sphinx-material):

<img width="1248" alt="sphinx_material" src="https://user-images.githubusercontent.com/350976/139913077-f3c5bdb5-0c99-4ec3-9765-456ce1a615d6.png">

[Pydata](https://github.com/pydata/pydata-sphinx-theme):

<img width="1224" alt="pydata" src="https://user-images.githubusercontent.com/350976/139913085-39ecc7e6-6b91-4008-b72c-1fcc4da95b67.png">

[Furo](https://github.com/pradyunsg/furo):

<img width="1157" alt="furo" src="https://user-images.githubusercontent.com/350976/139913090-c801939a-59c6-46d2-9f96-8a6a5b98407c.png">

And here's the original for reference, using the [Read the Docs theme](https://github.com/readthedocs/sphinx_rtd_theme):

<img width="1101" alt="read_the_docs" src="https://user-images.githubusercontent.com/350976/139913088-4829b482-001c-4c09-84e2-d767c83dc472.png">

Furo seems to be the best maintained, but people I've asked prefer the look of Sphinx Material slightly more.

The reason I originally chose Sphinx was the ability to auto generate API docs. I'm a big fan of this approach, as it means that people browsing the code have useful docstrings to read. MKDocs didn't historically support this, but it now seems possible via a [plugin](https://github.com/mkdocstrings/mkdocstrings).

Moving over to MKDocs would be quite a bit of work. I wouldn't even consider it, except that the [Material for MKDocs](https://squidfunk.github.io/mkdocs-material/) theme is so good. One approach might be to convert all of the Piccolo projects to use Sphinx Material, and try moving one of the smaller projects (like Piccolo Admin or Targ) to use MKDocs.

Does anyone have any thoughts on this? Should we change, or just keep what we have?

**Comment by sinisaos** [comment]
@dantownsend I also like the current theme, but if you plan to change the theme, I like [Furo](https://github.com/pradyunsg/furo) the most. I tried on the Piccolo API documentation and just changed this in `conf.py`
```python
html_theme = "furo"
pygments_style = "monokai"
pygments_dark_style = "monokai"
```
and result is very nice.

![Screenshot from 2021-11-02 19-18-12](https://user-images.githubusercontent.com/30960668/139923581-7e228a73-8011-4f62-80fc-42dad8e8175c.png)

**Reply by dantownsend** [reply]
Thanks - I didn't realise you could change the Pygments theme.

**Reply by dantownsend** [reply]
I had a play around with Furo, and it does look good like you suggested. Every time I went back to the current theme though, it just seemed right for some reason.  I've decided not to move from Sphinx. It's a minor pain sometimes, but it's so powerful.

So I'll keep with the current theme for now, and might move to Furo in the future.

I've been trying to tidy up the docs a bit, and adding a logo for each project.

There's now a logo for Piccolo admin:
https://github.com/piccolo-orm/piccolo_admin

And Targ:
https://github.com/piccolo-orm/targ

I was thinking something like this for Piccolo API.

<img width="669" alt="Screenshot 2021-11-03 at 22 09 16" src="https://user-images.githubusercontent.com/350976/140200155-164c6fc5-08d6-4e39-b6a5-cf0e4fa02976.png">

**Comment by dantownsend** [comment]
@sinisaos I've been on a journey of discovery this week, learning as much as possible about documentation and Sphinx.

The culmination is this custom Piccolo theme:

https://github.com/piccolo-orm/piccolo_theme

It will be used for all Piccolo projects moving forward.

It's just version 0.1 so we can continue to improve it, but I'm excited about it.

What do you think?

**Reply by sinisaos** [reply]
@dantownsend Looks great :+1:

**Reply by dantownsend** [reply]
Thanks!

**Comment by dantownsend** [comment]
The new theme is now live for Piccolo:

https://piccolo-orm.readthedocs.io/en/latest/index.html

Hopefully people like it!
